File: demo/safira.swift
struct Gemstone {
    let name: String
    let color: String
    let rarity: String
    
    // Method to describe the gemstone
    func describe() {
        print("The \(name) gemstone is a beautiful shade of \(color) and is \(rarity).")
    }
}

// Create a Gemstone instance
let myGemstone = Gemstone(name: "safira", color: "blue", rarity: "rare")

// Describe the gemstone
myGemstone.describe()

File: demo/safira.rs
struct Gemstone {
    name: String,
    color: String,
    rarity: String,
}

impl Gemstone {
    fn describe(&self) {
        println!("The {} gemstone is a beautiful shade of {} and is {}.", self.name, self.color, self.rarity);
    }
}

fn main() {
    // Create a Gemstone instance
    let my_gemstone = Gemstone {
        name: String::from("safira"),
        color: String::from("blue"),
        rarity: String::from("rare"),
    };

    // Describe the gemstone
    my_gemstone.describe();
}
